[AOTInductor] Add test for AOTInductorModel's interface

muchulee8 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #111325
* #111324

Summary:
Add a runner for AOTInductorModel for easier test.
Add test for existing AOTInductorModel's interface

Test Plan:
Commit itself is a test.

Reviewers:

Subscribers:

Tasks:

Tags:

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @aakhundov @ColinPeppler

muchulee8 committed:
[AOTInductor] Add test for AOTInductorModel's interface

Summary:
Add a runner for AOTInductorModel for easier test.
Add test for existing AOTInductorModel's interface (create & update constants map)

Test Plan:
Commit itself is a test.

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

desertfire commented on test/cpp/aot_inductor/test.cpp:
```diff
@@ -16,6 +17,65 @@
 namespace torch {
 namespace inductor {
 
+TEST(AotInductorModelTest, BasicTestCpu) {
```
Yang recently merged some refactoring to this file. You need to rebase to that.

desertfire commented on torch/csrc/inductor/aoti_model_runner.cpp:
```diff
@@ -0,0 +1,59 @@
+
+namespace torch::inductor {
+
+AOTIModelRunner::AOTIModelRunner(
```
Can we use some refactoring to share the code between AOTIModelRunner and AOTIModelContainerRunner?

muchulee8 replied:
I don't think it's that trivial after some look. It does seem like they are sharing lots of things, but in fact the types are all different, and since we're also loading from .so and using decltype for variables, it's not that easy to share/templatize.

muchulee8 committed:
Update on "[AOTInductor] Add test for AOTInductorModel's interface"


Summary:
Add a runner for AOTInductorModel for easier test.
Add test for existing AOTInductorModel's interface (create & update constants map)

Test Plan:
Commit itself is a test.

Reviewers:

Subscribers:

Tasks:

Tags:

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng aakhundov ColinPeppler

[ghstack-poisoned]
